src/sentry/snuba/discover.py

```diff
@@ -1112,10 +1120,40 @@ def find_measurements_min_max(measurements, min_value, max_value, user_query, pa
         min_value = min(min_values) if min_values else None
 
     if max_value is None:
-        max_values = [row[get_function_alias(column)] for column in max_columns]
+        max_values = [
+            row[get_function_alias("max(measurements.{})".format(measurement))]
+            for measurement in measurements
+        ]
         max_values = list(filter(lambda v: v is not None, max_values))
         max_value = max(max_values) if max_values else None
 
+        fences = []
+        if data_filter == "exclude_outliers":
+            for measurement in measurements:
+                # also known as the first quartile
+                p25_column = get_function_alias(
+                    "percentile(measurements.{}, 0.25)".format(measurement)
+                )
+                # also known as the third quartile
+                p75_column = get_function_alias(
+                    "percentile(measurements.{}, 0.75)".format(measurement)
+                )
+
+                first_quartile = row[p25_column]
+                third_quartile = row[p75_column]
+
+                if first_quartile is not None and third_quartile is not None:
+                    interquartile_range = abs(third_quartile - first_quartile)
+
+                    upper_outer_fence = third_quartile + 3 * interquartile_range
+                    fences.append(upper_outer_fence)
```

Comment:
Can you filter out `None`'s here? This can cause problems with the ongoing efforts to support python3 as `max([1, None])` behaves as expected in python2 but raise a `TypeError` in python3.

Reply:
I think `upper_outer_fence` guaranteed to never be None`.

Reply:
Ahh I see the check above here. All good 👍